File: BackEnd/src/main/java/com/example/BackEnd/DTO/DoctorDTO.java
package com.example.BackEnd.DTO;

import com.example.BackEnd.Entity.Department;
import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.*;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class DoctorDTO {
    @JsonProperty("department_id")
    private Department department;

    @JsonProperty("license_number")
    private String licenseNumber;

    @JsonProperty("experience_years")
    private Integer experienceYears;

    @JsonProperty("description")
    private String description;
}

File: FrontEnd/src/app/DTOs/doctor/doctor.DTO.ts
import {
    IsString,
} from 'class-validator';

export class DoctorDTO {
    @IsString()
    license_number: string;

    experience_years: number;

    @IsString()
    description: string;

    constructor(data: any) {
        this.license_number = data.license_number;
        this.experience_years = data.experience_years;
        this.description = data.description;
    }
}